File: backend/models/api/responses.py
"""
API response models
"""
from datetime import datetime
from typing import Any, Dict, List, Optional

from pydantic import BaseModel, Field

from ..enums import InquiryStatus, Priority, StoryCategory, StoryStatus
from ..schemas import StoryMetadata


class InquiryResponse(BaseModel):
    id: int
    user_id: str
    content: str
    language: str
    timestamp: datetime
    status: InquiryStatus
    metadata: Dict[str, Any]


class StoryResponse(BaseModel):
    id: int
    inquiry_id: int
    title: str
    description: str
    category: StoryCategory
    priority: Priority
    estimated_effort: float
    deadline: Optional[datetime] = None
    status: StoryStatus
    assignee: Optional[str] = None
    tags: List[str]
    dependencies: List[int]
    metadata: StoryMetadata
    created_at: datetime
    updated_at: datetime


class StoryGenerationResponse(BaseModel):
    inquiry_id: int
    stories: List[StoryResponse]
    status: str = Field(..., pattern="^(success|partial|failed)$")
    message: Optional[str] = None


class GenerationStatusResponse(BaseModel):
    inquiry_id: int
    status: str = Field(..., pattern="^(processing|completed|failed)$")
    progress: int = Field(..., ge=0, le=100)
    message: Optional[str] = None

File: frontend/src/types/api/responses.ts
/**
 * API response types
 */
import { InquiryStatus, StoryStatus, StoryCategory, Priority } from "../enums";
import { StoryMetadata } from "../models";

export interface InquiryResponse {
  id: number;
  userId: string;
  content: string;
  language: "ja" | "en";
  timestamp: string; // ISO string format for API
  status: InquiryStatus;
  metadata: {
    source: string;
    processingTime?: number;
    aiModel?: string;
  };
}

export interface StoryResponse {
  id: number;
  inquiryId: number;
  title: string;
  description: string;
  category: StoryCategory;
  priority: Priority;
  estimatedEffort: number;
  deadline?: string; // ISO string format for API
  status: StoryStatus;
  assignee?: string;
  tags: string[];
  dependencies: number[];
  metadata: StoryMetadata;
  createdAt: string; // ISO string format for API
  updatedAt: string; // ISO string format for API
}

export interface StoryGenerationResponse {
  inquiryId: number;
  stories: StoryResponse[];
  status: "success" | "partial" | "failed";
  message?: string;
}

export interface GenerationStatusResponse {
  inquiryId: number;
  status: "processing" | "completed" | "failed";
  progress: number; // 0-100
  message?: string;
}
